lib/Dialect/Torch/Transforms/AdjustCallingConventions.cpp

```diff
@@ -187,8 +188,49 @@ class AdjustCallingConventionForReturn
 };
 } // namespace
 
+static bool isValidNonContainerResultType(Type result_type) {
+  if (result_type.isa<Torch::BaseTensorType>() ||
+      result_type.isa<Torch::FloatType>() ||
+      result_type.isa<Torch::IntType>() ||
+      result_type.isa<Torch::BoolType>()) {
+    return true;
+  }
+  return false;
+}
+
+static LogicalResult validateReturns(func::FuncOp func) {
+  // getResultTypes can only ever be one item.  Multiple return values are
+  // stored in a tuple.
+  const auto& result_type = func.getResultTypes().front();
+
+  // Allow single tensor returns
```

Comment:
I've tried to update it.  It's quite verbose now, but more accurate.  Let me know if you have better ideas.  Happy to nitpick the phrasing to get it right, since it's what the user will see (for the error message).